### ISSUE TITLE
Locust Not Recognised as an Object

### ISSUE THREAD

**Issue by bernmang** [subhacom/argos#3]
I can play the video within the GUI but when I click on it, it collapses. I am using a Windows system and assume it is not automatically converting the float into an integer. 

The lines of code in which this has occurred:
Traceback (most recent call last):
  File "C:\Users\Berni\anaconda3\envs\track\lib\site-packages\argos\frameview.py", line 569, in setRectangles
    self._updateItemDisplay()
  File "C:\Users\Berni\anaconda3\envs\track\lib\site-packages\argos\frameview.py", line 401, in _updateItemDisplay
    pen.setWidth(self.linewidth)
TypeError: setWidth(self, int): argument 1 has unexpected type 'float'

There is no recognition of the locust in my videos as an object, despite having trained a neural network. In looking at the code with my supervisor, we saw there was mention of bbox and were wondering if it has to do with this code above and that's why no bounding box is being displayed. Another possibility we were thinking of is that it is not an error within frameview.py but somewhere else that may not be showing up within Anaconda Prompt.

**Comment by subhacom** [comment]
Thanks for reporting this. The exception message is spot on, and again it is a type compatibility issue. I have fixed it and uploaded a new version of argos on PyPI. Can you please try installing this like before, either using either

```
pip install git+https://github.com/subhacom/argos.git@Fix_1_playtimer
```

or just

```
pip install argos
```
and see if that fixes the issue?

**Comment by bernmang** [comment]
The locust can now be recognised as an object when I use the classical segmentation but still nothing with the YOLACT segmentation method. This is most likely an issue with my configuration and/or weights file though.

**Comment by subhacom** [comment]
Sorry to hear that. Did you try the sample configuration and weights for locust hatchlings? If your video background is relatively uncluttered, this might already work for you.
The files are available here: https://www.dropbox.com/sh/9fcgouv6wsjvypk/AAC5A2BIrbpdOG5vy8YwOk6ca?dl=0

**Comment by bernmang** [comment]
I have now tried the sample configuration and weights files but they do not seem to be working either.

**Comment by subhacom** [comment]
Do you get any other exceptions on the command line? Can you please share
the output on the command line? Also, if possible please share (on some
online file-sharing system) one of your video files, so I can check it
locally.

**Comment by subhacom** [comment]
I am closing this issue, since the reported error has been resolved. Please open another one if you are facing another problem.

**Comment by bernmang** [comment]
There don't usually seem to be any exceptions on the command line whilst it
is running. I did let the programme run in the background once and got the
following error when I looked back, unsure if it is relevant to the issue
though:
Traceback (most recent call last):
File
"C:\Users\Berni\anaconda3\envs\track\lib\site-packages\argos\csrtracker.py",
line 129, in track
    valid = self.find_nonoverlapping(bboxes)
  File
"C:\Users\Berni\anaconda3\envs\track\lib\site-packages\argos\csrtracker.py",
line 83, in find_nonoverlapping
    dist = pairwise_distance(bboxes, bboxes,
  File "argos\cutility.pyx", line 173, in argos.cutility.pairwise_distance
ValueError: Buffer has wrong number of dimensions (expected 2, got 1)

This link should direct you to a few of the videos I am using -
https://drive.google.com/drive/folders/1VNJIpw8e7stZRtz31tTBRfGGpgpo4lvV?usp=share_link
.

L.greg-0001-L.mp4 was the video file used for the training. I did try both
the yolact_base and baby_locust configuration and weights file. Nothing was
recognised with the yolact_base configuration and weights file but when the
baby_locust configuration and weights files were used on the
L.greg-0002-L.mp4 video file, there was recognition of something in the
arena as an object, although it was not the locust.

**Comment by subhacom** [comment]
From that error it looks like you are using the CSRT tracking. I am getting a perfect tracking  with SORT for the first video when using the default weights trained on locusts. In my experience CSRT is not very reliable over longer periods of time. However I shall track down and fix this bug.

**Comment by subhacom** [comment]
This should be fixed in the latest bug-fix version: ` argos-toolkit-0.1.3.post3`. Please try installing the same from pypi.

**Comment by bernmang** [comment]
Great, thank you so much for your help!
